### PR TITLE
fix(HMSPROV-414): start dequeue loop in api only for memory

### PR DESCRIPTION
**Pull request by lzap**
So the job queue is designed to run in two modes. When memory is configured, the pbapi (api worker) should automatically start goroutine that is responsible for performing background jobs. The communication is through local memory queue, this is great for development and it is the default mode when you don't provide any setting.

When in redis mode, the api worker was not supposed to start any goroutines, however, due to overlook it started handlers but did not register any functions. This meant that in stage, 4 workers are started (2 pbworkers, 2 pbapi) but two of them (pbapi) did not have registered any handler functions. Meaning the jobs were silently dropped, this was a bug in dejq redis implementation which has been fixed (error is now logged in that case).

This patch fixes the problem, it is a one-liner, however while working on that I improved dejq API by adding a job UUID. This now greatly increases logging and searching within job messages.

I am also increasing replicas back to 2/2 (api/workers) since decreasing to 1 did no do anything in kubernetes, I do not know how to decrease it at the moment and cloudwatch/kibana logging is now ready.

### CODE REVIEW

File: internal/config/helpers.go
Comment (lzap) [inline internal/config/helpers.go:86]: I want the binary name to be reported in logs because that is our cloudwatch stream name.

File: internal/jobs/queue/queue_interface.go
Comment (lzap) [inline internal/jobs/queue/queue_interface.go:11]: This change is needed because dejq v1.4 changed its API. And yes, I did a breaking change in a minor release, there are no other users of dejq than me :-)

File: deploy/clowdapp.yaml
Comment (lzap) [inline deploy/clowdapp.yaml:263]: I noticed this, this is now unused. Cloudwatch is configured directly from clowder.

**Review by ezr-ondrej — APPROVED** [review]
Needs rebase but apart of it LGTM :+1:

**Comment by ezr-ondrej** [comment]
On related note, but just a thought not blocking the PR: if we are keeping the `dejq` I'd vote for merging it in the repository. Given we are the only user, it would at least be clear that this codebase relies on it and we would keep track of changes here. I guess that keeping changelong of external library for no reason makes no sense and it could also get more eyes. Just a thought tho :)

**Comment by lzap** [comment]
I thought about this too, makes sense tho I would propose to postpone this until I have more time. I would probably drop dbjobqueue as well and simplify the interface too.

**Comment by lzap** [comment]
What? I did not hit the merge button!!! WTF

**Comment by lzap** [comment]
GH automatically merged this after I rebased the PR while it was still running tests!

Snímek obrazovky 2023-01-12 v 8 29 18

**Pull request merged**
